src/storage.zig

```diff
@@ -315,3 +375,52 @@ test "deleteBookmark" {
     try testing.expectEqualStrings("gh", results[0].value);
     try testing.expectEqualStrings("hn", results[1].value);
 }
+
+test "from_csv test malformed data" {
+    const data =
+        \\gh
+        \\
+    ;
+    var reader = std.Io.Reader.fixed(data);
+    const results = try Bookmark.from_csv(testing.allocator, &reader);
+    defer testing.allocator.free(results.bookmarks);
+    try testing.expectEqual(1, results.failed_count);
+}
+
+test "from_csv" {
+    const data =
+        \\gh,https://www.github.com,dev,code,
+        \\gl,https://gitlab.com,dev,ci,vcs,
+        \\
+    ;
+    var reader = std.Io.Reader.fixed(data);
+    const results = try Bookmark.from_csv(testing.allocator, &reader);
```

Comment:
Just to reiterate on the point made about the `from_csv` file, I think this line helps illustrate my feelings: Bookmark.from_csv returning not a bookmark, not a list of bookmarks (which I think would still be wrong in this case), but a response type indicating how the import went makes me feel like this deserves it's own organizational pattern (perhaps a BookmarkImporter struct, but I'm open to other ideas, too!)